store/src/main/java/Store.java

```diff
@@ -0,0 +1,112 @@
+                .sorted(nameComparator).collect(Collectors.toList());
+        return sortedByName;
+    }
+
+    public ArrayList<Product> sortByNameDesc() {
+        ProductNameComparator nameComparator = new ProductNameComparator();
+        ArrayList<Product> sortedByName = (ArrayList<Product>) products.stream()
+                .sorted(nameComparator.reversed()).collect(Collectors.toList());
+        return sortedByName;
+    }
+
+
+    public ArrayList<Product> sortByPriceAsc() {
+        ProductPriceComparator priceComparator = new ProductPriceComparator();
+        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
+                .sorted(priceComparator).collect(Collectors.toList());
+        return sortedByPrice;
+    }
+
+    public ArrayList<Product> sortByPriceDesc() {
+        ProductPriceComparator priceComparator = new ProductPriceComparator();
+        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
+                .sorted(priceComparator.reversed()).collect(Collectors.toList());
+        return sortedByPrice;
+    }
+
+
+    public ArrayList<Product> sortByRatingAsc() {
+        ProductRatingComparator ratingComparator = new ProductRatingComparator();
+        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
+                .sorted(ratingComparator).collect(Collectors.toList());
+        return sortedByPrice;
+    }
+
+    public ArrayList<Product> sortByRatingDesc() {
+        ProductRatingComparator ratingComparator = new ProductRatingComparator();
+        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
+                .sorted(ratingComparator.reversed()).collect(Collectors.toList());
+        return sortedByPrice;
+    }
+
+
+    public ArrayList<Product> sortProduct(Map<String, String> map, String sortingKey) {
+        ArrayList<Product> sortedListOfProducts = null;
+        String value = map.get(sortingKey);
+        switch (sortingKey) {
+            case "name":
+                if (value.equals("asc")) {
+                    sortedListOfProducts = sortByNameAsc();
+                } else if (value.equals("desc")) {
+                    sortedListOfProducts = sortByNameDesc();
+                }
+                break;
+            case "price":
+                if (value.equals("asc")) {
+                    sortedListOfProducts = sortByPriceAsc();
+                } else if (value.equals("desc")) {
+                    sortedListOfProducts = sortByPriceDesc();
+                }
+                break;
+            case "rate":
+                if (value.equals("asc")) {
+                    sortedListOfProducts = sortByRatingAsc();
+                } else if (value.equals("desc")) {
+                    sortedListOfProducts = sortByRatingDesc();
+                }
+                break;
+        }
+
+        return sortedListOfProducts;
+    }
+
+    public void showTopProductsByPriceDesc(int n) {
```

Comment:
please read aboud SOLID principle, and split this method in several, and method that get smthing from user (Scanner) should not be in store class